src/vec_int32_ppc.h

```diff
@@ -0,0 +1,649 @@
+  wmask01 = vec_vmrghw (zero, ones);
+
+  mt = (vui16_t)b;
+  mtl = vec_mergel (mt, mt);
+  mth = vec_mergeh (mt, mt);
+#ifdef _ARCH_PWR7
+  m0 = (vui16_t)vec_xxpermdi ((vui64_t)mth, (vui64_t)mtl, 3);
+#else
+  {
+    vui32_t temp;
+    temp = vec_sld ((vui32_t) mtl, (vui32_t) mtl, 8);
+    result = (vui64_t) vec_sld ((vui32_t) mth, temp, 8);
+  }
+#endif
+
+  resw = vec_sld (a, a, 4);
+  m1 = (vui16_t)vec_sel (resw, a, wmask01);
+
+  p0 = vec_vmuleuh (m1, m0);
+  p1 = vec_vmulouh (m1, m0);
+
+  resw = vec_sel (p0, p1, wmask01);
+  res = (vui64_t)resw;
+
+  pp10 = vec_sld (p1, p1, 12);
+  pp01 = p0;
+  /* pp01 = vec_addudm (pp01, pp10).  */
+  {
+    vui32_t c;
+    vui32_t xmask;
+    xmask = vec_sld (wmask01, wmask01, 2);
+    c    = vec_vaddcuw (pp01, pp10);
+    pp01 = vec_vadduwm (pp01, pp10);
+    c    = vec_sld (c, c, 6);
+    pp01 = vec_sld (pp01, pp01, 2);
+    pp01 = vec_sel (c, pp01, xmask);
+  }
+  /* res = vec_addudm (pp01, res).  */
+  {
+    vui32_t c, r;
+    c = vec_vaddcuw (pp01, (vui32_t)res);
+    r = vec_vadduwm (pp01, (vui32_t)res);
+    c = vec_sld (c, zero, 4);
+    res = (vui64_t)vec_vadduwm (r, c);
+  }
+#endif
+  return (res);
+}
+
+/** \brief Vector Multiply Unsigned Word Modulo.
+ *
+ *  Multiple the corresponding word elements of two vector unsigned int
+ *  values and return the low order 32-bits of the 64-bit product for
+ *  each element.
+ *
+ *  @param a 128-bit vector signed int.
+ *  @param b 128-bit vector signed int.
+ *  @return vector low order 32-bit of the product of the word elements
+ *  of a and b.
+ */
+static inline vui32_t
+vec_muluwm (vui32_t a, vui32_t b)
+{
+#if __GNUC__ >= 7
+  return vec_mul (a, b);
+#else
+  vui32_t r;
+#ifdef _ARCH_PWR8
+  __asm__(
+      "vmuluwm %0,%1,%2;\n"
+      : "=v" (r)
+      : "v" (a),
+      "v" (b)
+      : );
+#else
+  vui32_t s16 = (vui32_t)vec_vspltisw (-16);
+  vui32_t z = (vui32_t)vec_vspltisw (0);
+  vui32_t t4;
+  vui32_t t2, t3;
+  vui16_t t1;
+
+  t1 = (vui16_t)vec_vrlw (b, s16);
+  t2 = vec_vmulouh ((vui16_t)a, (vui16_t)b);
+  t3 = vec_vmsumuhm ((vui16_t)a, t1, z);
+  t4 = vec_vslw (t3, s16);
+  r = (vui32_t)vec_vadduwm (t4, t2);
+#endif
+  return (r);
+#endif
+}
+
+/** \brief Vector Population Count word.
+ *
+ *  Count the number of '1' bits (0-32) within each word element of
+ *  a 128-bit vector.
+ *
+ *  For Power8 (PowerISA 2.07B) or later use the Vector Population
+ *  Count Word instruction. Otherwise use the pveclib vec_popcntb
+ *  to count each byte then sum across with Vector Sum across Quarter
+ *  Unsigned Byte Saturate.
+ *
+ *  @param vra 128-bit vector treated as 4 x 32-bit integer (words)
+ *  elements.
+ *  @return 128-bit vector with the population count for each word
+ *  element.
+ */
+#ifndef vec_popcntw
+static inline vui32_t
+vec_popcntw (vui32_t vra)
+{
+  vui32_t r;
+#ifdef _ARCH_PWR8
+#ifndef vec_vpopcntw
+  __asm__(
+      "vpopcntw %0,%1;"
+      : "=v" (t)
+      : "v" (vra)
+      : );
```

Comment:
`r` is not set here, but is returned later.  Oh, maybe that `t` is supposed to be `r`?